### PR TITLE
[AMS-656] checking for existing locks while list versions

### PR DESCRIPTION
**Pull request by denisivan0v**

### CODE REVIEW

File: src/VStore/Templates/TemplatesStorageReader.cs [hunk 1]
```diff
@@ -121,6 +124,8 @@ public TemplatesStorageReader(CephOptions cephOptions, IAmazonS3Proxy amazonS3)
 
         public async Task<string> GetTemplateLatestVersion(long id)
         {
+            await _lockSessionManager.EnsureLockSessionNotExists(id);
```
Comment (xakep139) [inline src/VStore/Templates/TemplatesStorageReader.cs]: I don't think it's necessary to check lock session on reading

File: src/VStore/Objects/ObjectsStorageReader.cs [hunk 2]
```diff
@@ -215,6 +221,8 @@ async Task<(bool IsTruncated, int NextVersionIndex, string NextVersionIdMarker)>
 
         public async Task<IReadOnlyCollection<VersionedObjectDescriptor<string>>> GetObjectLatestVersions(long id)
         {
+            await _lockSessionManager.EnsureLockSessionNotExists(id);
+
             var versionsResponse = await _amazonS3.ListVersionsAsync(_bucketName, id + "/");
```
Comment (xakep139) [inline src/VStore/Objects/ObjectsStorageReader.cs]: Object key should ends with `object` postfix, right?

File: src/VStore/Templates/TemplatesStorageReader.cs (hunk 1 above)
Comment (xakep139) [inline src/VStore/Templates/TemplatesStorageReader.cs]: Besides you can check implementation of `TemplatesManagementService.CreateTemplate()` - call to `_templatesStorageReader.GetTemplateLatestVersion()` would always throw an exception due to existing lock session

Comment (xakep139) [inline src/VStore/Templates/TemplatesStorageReader.cs]: Same problem would affect `ObjectsManagementService.PutObject()`

File: src/VStore/Objects/ObjectsStorageReader.cs [hunk 3]
```diff
@@ -72,12 +72,6 @@ public sealed class ObjectsStorageReader
                                     ObjectMetadataRecord record;
                                     try
                                     {
-                                        var response = await _amazonS3.GetObjectMetadataAsync(_bucketName, id.AsS3ObjectKey(Tokens.ObjectPostfix));
-                                        var metadataWrapper = MetadataCollectionWrapper.For(response.Metadata);
-                                        var author = metadataWrapper.Read<string>(MetadataElement.Author);
-                                        var authorLogin = metadataWrapper.Read<string>(MetadataElement.AuthorLogin);
-                                        var authorName = metadataWrapper.Read<string>(MetadataElement.AuthorName);
-
                                         var objectVersions = await GetObjectLatestVersions(id);
```
Comment (xakep139) [inline src/VStore/Objects/ObjectsStorageReader.cs]: Get latest version first and then fetch metadata

File: src/VStore/Templates/TemplatesStorageReader.cs [hunk 4]
```diff
@@ -58,13 +58,12 @@ public TemplatesStorageReader(CephOptions cephOptions, IAmazonS3Proxy amazonS3,
                                     ObjectMetadataRecord record;
                                     try
                                     {
-                                        var response = await _amazonS3.GetObjectMetadataAsync(_bucketName, templateId.ToString());
+                                        var versionId = await GetTemplateLatestVersion(templateId);
```
Comment (xakep139) [inline src/VStore/Templates/TemplatesStorageReader.cs]: Get latest version first and then fetch metadata

File: src/VStore/Templates/TemplatesStorageReader.cs (hunk 1 above)
Comment (denisivan0v) [inline src/VStore/Templates/TemplatesStorageReader.cs]: LGMT, thanks!

**Pull request merged**